snippets/csharp/System.Net/HttpVersion/Overview/httpversion_version10.cs

```diff
@@ -18,28 +18,31 @@ public static void Main()
 		try	
 		{
 // <Snippet1>
-			// Create a 'HttpWebRequest' object.
-			HttpWebRequest myHttpWebRequest=(HttpWebRequest)WebRequest.Create("http://www.microsoft.com");
-			Console.WriteLine("\nThe 'ProtocolVersion' of the protocol before assignment is :{0}",myHttpWebRequest.ProtocolVersion);
-			// Assign Version10 to ProtocolVersion.
-			myHttpWebRequest.ProtocolVersion=HttpVersion.Version10;
-			// Assign the response object of 'HttpWebRequest' to a 'HttpWebResponse' variable.
-			HttpWebResponse myHttpWebResponse=(HttpWebResponse)myHttpWebRequest.GetResponse();
-			Console.WriteLine("\nThe 'ProtocolVersion' of the protocol after  assignment is :{0}",myHttpWebRequest.ProtocolVersion);
-			Console.WriteLine("\nThe 'ProtocolVersion' of the response object is :{0}",myHttpWebResponse.ProtocolVersion);
+			// HttpClient lifecycle management best practices:
+			// https://learn.microsoft.com/dotnet/fundamentals/networking/http/httpclient-guidelines#recommended-use
+			HttpClient client = new HttpClient();
+
+			HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, "http://www.microsoft.com");
+			Console.WriteLine("Default HTTP request version is {0}", request.Version);
+
+			request.Version = HttpVersion.Version10;
+			Console.WriteLine("Request version after assignment is {0}", request.Version);
+
+			HttpResponseMessage response = client.Send(request);
+			Console.WriteLine("Response HTTP version {0}", response.Version);
```

Comment:
The HttpClient instead and the HttpResponseMessage should both be disposed / should both be in using blocks / declarations.